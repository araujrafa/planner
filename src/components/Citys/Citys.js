import React from 'react';

export default ({ citys, onChangeCity }) => (
  <select onChange={onChangeCity}>
    <option select="true" value=''>Selecione uma cidade...</option>
    {citys.map(city => {
      return <option key={city.woeid} value={city.woeid}>{city.district}</option>
    })}
  </select>
)
