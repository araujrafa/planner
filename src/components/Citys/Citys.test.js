import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';
import Citys from './Citys';

describe('Citys component', () => {
  const props = {
    citys: [{district: 'SÃ£o Paulo', woeid: '1'}, {district: 'Rio de Janeiro', woeid: '2'}],
    onChangeCity: jest.fn(),
    label: 'Escolha a cidade'
  }

  test('Citys changes the class when hovered', () => {  
    const component = renderer.create(
      <Citys {...props}/>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('Citys call funtion after change', () => {
    const wrapper = shallow(<Citys {...props}/>);
  
    wrapper.find('select').simulate('change');
  
    expect(props.onChangeCity).toHaveBeenCalledTimes(1);
  });
})