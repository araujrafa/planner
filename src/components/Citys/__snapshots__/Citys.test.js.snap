// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Citys component Citys changes the class when hovered 1`] = `
<div
  className="form-group"
>
  <label>
    Escolha a cidade
  </label>
  <select
    className="form-control"
    onChange={[Function]}
  >
    <option
      select="true"
      value=""
    >
      Selecione uma cidade...
    </option>
    <option
      value="1"
    >
      SÃ£o Paulo
    </option>
    <option
      value="2"
    >
      Rio de Janeiro
    </option>
  </select>
</div>
`;
